log switch_format, mipi_syst tests failing on intel_adsp_ace15_mtpm

**Describe the bug**
These testcases fail to build using xtensa toolchain on intel_adsp_ace15_mtpm platform.

```
tests/subsys/logging/log_switch_format/logging.log_switch_format.deferred
tests/subsys/logging/log_switch_format/logging.log_switch_format.immediate
tests/subsys/logging/log_switch_format/logging.log_switch_format.custom_output
tests/subsys/logging/log_syst/logging.log_syst.mipi_syst
```

**To Reproduce**
west build -p -b intel_adsp_ace15_mtpm tests/subsys/logging/log_switch_format/

**Expected behavior**
Testcase needs to pass

**Impact**
Failing testcase

**Logs and console output**
```
In file included from zephyr/lib/libc/newlib/libc-hooks.c:12:
zephyr/include/zephyr/posix/sys/stat.h:40:10: fatal error: 'sys/_timespec.h' file not found
#include <sys/_timespec.h>
         ^~~~~~~~~~~~~~~~~
1 error generated.
```

**Environment (please complete the following information):**
 - OS:  Linux
 - Toolchain: RI-2021.7
 - Commit SHA: 461a13c085de73aabdb5777247a96675472a73ad

--- a/lib/libc/Kconfig
+++ b/lib/libc/Kconfig
@@ -19,7 +19,7 @@ config SUPPORT_MINIMAL_LIBC
 config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
-	depends on "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "arcmwdt"
+	depends on "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
 	depends on !(CPP && "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr")
 	# picolibc text is outside .pinned.text on this board. #54148
 	default y

